### PR TITLE
Adds support for force uploading packages

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/src/depot/server.rs [hunk 1]
```diff
@@ -829,9 +842,17 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
 
     // Return conflict only if we have BOTH package metadata and a valid
     // archive on disk.
+    let upload_forced = check_forced(req);
     let origin_package_found =
         match route_message::<OriginPackageGet, OriginPackage>(req, &ident_req) {
-            Ok(_) => return Ok(Response::with(status::Conflict)),
+            Ok(_) => {
+                if upload_forced {
+                    debug!("An upload was forced, bypassing database validation!");
```
Comment (chefsalim) [inline components/builder-api/src/depot/server.rs]: Might be good to add the ident of the forced upload into the debug msg..

File: components/builder-api/src/depot/server.rs [hunk 2]
```diff
@@ -428,6 +428,19 @@ pub fn origin_member_delete(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn check_forced(req: &mut Request) -> bool {
+    match helpers::extract_query_value("forced", req) {
```
Comment (chefsalim) [inline components/builder-api/src/depot/server.rs]: could probably simplify this function to:
```
if let Some(flag) = helpers::extract_query_value("forced", req) {
  flag == "true"
} else {
  false
}
```

Comment (eeyun) [reply components/builder-api/src/depot/server.rs]: Ah clever!

**Pull request merged**